pkg/aws/cloud.go

```diff
@@ -200,14 +201,18 @@ func inferVPCID(ec2Metadata services.EC2Metadata, ec2Service services.EC2) (stri
 	return "", amerrors.NewAggregate(errList)
 }
 
-func inferVPCIDFromTags(ec2Service services.EC2, VpcNameTagKey string, VpcNameTagValue string) (string, error) {
+func inferVPCIDFromTags(ec2Service services.EC2, VpcTags map[string]string) (string, error) {
+	vpcFilter := []ec2types.Filter{}
+
+	for tagKey, tagValue := range VpcTags {
+		vpcFilter = append(vpcFilter, ec2types.Filter{
+			Name:   aws.String(fmt.Sprintf("tag:%s", tagKey)),
+			Values: []string{tagValue},
+		})
+	}
+
 	vpcs, err := ec2Service.DescribeVPCsAsList(context.Background(), &ec2.DescribeVpcsInput{
-		Filters: []ec2types.Filter{
-			{
-				Name:   aws.String("tag:" + VpcNameTagKey),
-				Values: []string{VpcNameTagValue},
-			},
-		},
+		Filters: vpcFilter,
 	})
 	if err != nil {
 		return "", fmt.Errorf("failed to fetch VPC ID with tag: %w", err)
```

Comment:
I think this should say "... with tags:" or ".. with tag(s):" now. Same for the errors below.